### PR TITLE
Add 2023 New Mexico Tax Rules

### LINKED ISSUE #4192: Add 2023 New Mexico Tax Rules
2023 Income Tax Form Instructions: https://klvg4oyd4j.execute-api.us-west-2.amazonaws.com/prod/PublicFiles/34821a9573ca43e7b06dfad20f5183fd/90560f4e-0ef0-4e52-a003-878b84f858bb/PITbook2023.pdf

### PR DESCRIPTION
**Pull request by xianxinzhu**
Fixes #4192

### CODE REVIEW

File: policyengine_us/variables/gov/states/de/tax/income/combined_separate_status/de_personal_credit.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/de/tax/income/combined_separate_status/de_personal_credit.py:1]: I dont think this should be in this PR right?

Comment (xianxinzhu) [reply policyengine_us/variables/gov/states/de/tax/income/combined_separate_status/de_personal_credit.py:1]: Yes, I'll delete that. I actually have no clue why it is here. Sorry

**Comment by PavelMakarchuk** [comment]
Could you sum up the changes that happened here? It is easier to double check

**Comment by xianxinzhu** [comment]
- Update the low income comprehensive tax rebate for filers with different exemptions.
- Add the 2023 references for the unchanging parameters.

File: changelog_entry.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - Add 2023 New Mexico Tax Rules.
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
    - 2023 New Mexico Tax Rules.
```

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/parameters/gov/states/nm/tax/income/rebates/low_income/amount/five_exemptions.yaml [hunk 2]
```diff
@@ -4,78 +4,97 @@ brackets:
       2021-01-01: 0
     amount:
       2021-01-01: 455
+      2023-01-01: 492
   - threshold: 
       2021-01-01: 1_001
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/nm/tax/income/rebates/low_income/amount/five_exemptions.yaml:9]: Then use `calc(..., right=True)` in the formula. This better mirrors the law.
```suggestion
      2021-01-01: 1_000
```

File: policyengine_us/parameters/gov/states/nm/tax/income/rebates/low_income/amount/five_exemptions.yaml [hunk 3]
```diff
@@ -86,6 +105,8 @@ metadata:
   type: single_amount
   label: New Mexico comprehensive low income tax rebate five exemptions amount
   reference:
+    - title: New Mexico 2023 PIT-RC, page 82, SECTION II LOW INCOME COMPREHENSIVE TAX REBATE
+      href: https://klvg4oyd4j.execute-api.us-west-2.amazonaws.com/prod/PublicFiles/34821a9573ca43e7b06dfad20f5183fd/90560f4e-0ef0-4e52-a003-878b84f858bb/PITbook2023.pdf#page=82
     - title: New Mexico 2022 PIT-RC, page 58, SECTION II LOW INCOME COMPREHENSIVE TAX REBATE
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/nm/tax/income/rebates/low_income/amount/five_exemptions.yaml:110]:
```suggestion
    # The law adjusts for inflation beginning in 2022, but no values changed in 2022.
    - title: New Mexico 2022 PIT-RC, page 58, SECTION II LOW INCOME COMPREHENSIVE TAX REBATE
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/nm/tax/income/rebates/low_income_comprehensive/nm_low_income_comprehensive_tax_rebate.py [hunk 4]
```diff
@@ -35,12 +35,12 @@ def formula(tax_unit, period, parameters):
                 exemptions >= 6,
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/nm/tax/income/rebates/low_income_comprehensive/nm_low_income_comprehensive_tax_rebate.py:35]: delete and use `default`

File: policyengine_us/variables/gov/states/nm/tax/income/rebates/low_income_comprehensive/nm_low_income_comprehensive_tax_rebate.py [hunk 5]
```diff
@@ -35,12 +35,12 @@ def formula(tax_unit, period, parameters):
...
-                p.amount.three_exemptions.calc(agi),
-                p.amount.four_exemptions.calc(agi),
-                p.amount.five_exemptions.calc(agi),
-                p.amount.six_exemptions.calc(agi),
+                p.amount.one_exemption.calc(agi, right=True),
+                p.amount.two_exemptions.calc(agi, right=True),
+                p.amount.three_exemptions.calc(agi, right=True),
+                p.amount.four_exemptions.calc(agi, right=True),
+                p.amount.five_exemptions.calc(agi, right=True),
+                p.amount.six_exemptions.calc(agi, right=True),
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/nm/tax/income/rebates/low_income_comprehensive/nm_low_income_comprehensive_tax_rebate.py:43]:
```suggestion
], default = p.amount.six_exemptions.calc(agi, right=True),
```

**Review by MaxGhenis — APPROVED** [review]
good to merge after this

**Pull request merged**